FeedSX/customModals/ReportModal/index.tsx

```diff
@@ -0,0 +1,259 @@
+import {
+  View,
+  Text,
+  Modal,
+  Pressable,
+  TouchableOpacity,
+  TextInput,
+  Image,
+} from 'react-native';
+import React, {useEffect, useState} from 'react';
+import {PostUI} from '../../Models/PostModel';
+import styles from './styles';
+import {useDispatch} from 'react-redux';
+import {getReportTags, postReport} from '../../store/actions/feed';
+import {GetReportTagsRequest, PostReportRequest} from 'likeminds-sdk';
+import {useAppSelector} from '../../store/store';
+import {LMLoader} from '../../components';
+import {
+  COMMENT_REPORT_ENTITY_TYPE,
+  COMMENT_TYPE,
+  POST_REPORT_ENTITY_TYPE,
+  POST_TYPE,
+  REPLY_REPORT_ENTITY_TYPE,
+} from '../../constants/Strings';
+import {showToastMessage} from '../../store/actions/toast';
+
+// interface for post report api request
+interface ReportRequest {
+  entityId: string;
+  entityType: number;
+  reason?: string;
+  tagId: number;
+  uuid: string;
+}
+
+// post report modal props
+interface ReportModalProps {
+  visible: boolean;
+  closeModal: () => void;
+  reportType: string;
+}
```

Comment:
why we have created this here and what is the need for this?

Reply:
We need data from two interfaces over here so this interface defines the props that this modal expects for handling its visibilty and the content in the modal is shown from the PostUI interface, which the 2nd interface.